app/contracts/attachments/create_contract.rb

```diff
@@ -50,7 +50,7 @@ class CreateContract < ::ModelContract
     def validate_attachments_addable
       return if model.container
 
-      if Redmine::Acts::Attachable.attachables.none?(&:attachments_addable?)
+      if Redmine::Acts::Attachable.attachables.none?(&:uncontainered_attachable?)
```

Comment:
@oliverguenther Shouldn't this check for `allow_uncontainered` depend on whether or not we have an uncontainered attachment here? If the container attribute is set I would expect the contract to work even if the option was false.

Reply:
You're looking at the validation for uncontainered attachments. cf., line 50, it does not access this block if there is a container present

Reply:
This is what happens when you don't even look at the context one line above! Got it thanks.

All good then!